src/domain/tradingview/useTVDatafeed.ts

```diff
@@ -1,20 +1,29 @@
-import { HistoryCallback, PeriodParams, ResolutionString, SubscribeBarsCallback } from "charting_library";
+import {
+  DatafeedConfiguration,
+  HistoryCallback,
+  IDatafeedChartApi,
+  IExternalDatafeed,
+  LibrarySymbolInfo,
+  PeriodParams,
+  ResolutionString,
+  SubscribeBarsCallback,
+} from "charting_library";
 import { getNativeToken, getPriceDecimals, getTokens, isChartAvailabeForToken } from "config/tokens";
+import { SUPPORTED_RESOLUTIONS_V1 } from "config/tradingview";
 import { useChainId } from "lib/chains";
-import { useEffect, useMemo, useRef } from "react";
+import { MutableRefObject, useEffect, useMemo, useRef } from "react";
 import { TVDataProvider } from "./TVDataProvider";
-import { Bar, SymbolInfo } from "./types";
+import { Bar, FromOldToNewArray, SymbolInfo } from "./types";
 import { formatTimeInBarToMs } from "./utils";
-import { SUPPORTED_RESOLUTIONS_V1 } from "config/tradingview";
 
-function getConfigurationData(supportedResolutions) {
+function getConfigurationData(supportedResolutions): DatafeedConfiguration {
   return {
     supported_resolutions: Object.keys(supportedResolutions),
     supports_marks: false,
     supports_timescale_marks: false,
     supports_time: true,
     reset_cache_timeout: 100,
-  };
+  } as unknown as DatafeedConfiguration;
```

Comment:
let's do rather smth like
<img width="819" alt="image" src="https://github.com/gmx-io/gmx-interface/assets/144131435/c82a30e0-b055-4aef-972b-f3f35026ada9">

also `ResolutionString` looks weird type, thinking if we could just use `string` instead?
And reset_cache_timeout is missing in `DatafeedConfiguration`
<img width="442" alt="image" src="https://github.com/gmx-io/gmx-interface/assets/144131435/d1ba1cde-b491-4e5e-b10b-4a6d67219f5e">

Reply:
1) fixed
2) added to ts